core/chain-configs/src/client_config.rs

```diff
@@ -167,40 +167,65 @@ pub struct StateSplitConfig {
     /// The soft limit on the size of a single batch. The batch size can be
     /// decreased if resharding is consuming too many resources and interfering
     /// with regular node operation.
+    #[serde(default = "default_batch_size")]
```

Comment:
I'm wondering if we can use the serde default attribute for structs. I haven't tested it but it's behavior should be different from the serde default attribute of fields, where for structs, we take the field default value from Default::default()

Documentation here: https://serde.rs/container-attrs.html#default

Reply:
Let me test it out.

Reply:
Just posted in comment, good timings